src/ui/public/vislib/lib/axis.js

```diff
@@ -247,21 +256,48 @@ export default function AxisFactory(Private) {
           const width = parentWidth / n;
           const height = $(this.parentElement).height();
 
-          self.validateWidthandHeight(width, height);
+          /*
+           const width = $(el).parent().width();
+           const height = $(el).height();
+           */
+          const adjustedHeight = height - margin.top - margin.bottom;
 
-          self.getXAxis(width);
+
+          self.validateWidthandHeight(width, height);
 
           const svg = div.append('svg')
           .attr('width', width)
           .attr('height', height);
 
-          svg.append('g')
-          .attr('class', 'x axis')
-          .attr('transform', 'translate(0,0)')
-          .call(self.xAxis);
+          if (self.type === 'category') {
```

Comment:
as in the following PRs i would make both axis to be more alike this IF statement will go away.